app/helpers/foreman_resource_quota/resource_quota_helper.rb

```diff
@@ -2,40 +2,110 @@
 
 module ForemanResourceQuota
   module ResourceQuotaHelper
+    FACTOR_B_TO_KB = 1024
     FACTOR_B_TO_MB = 1024 * 1024
     FACTOR_B_TO_GB = 1024 * FACTOR_B_TO_MB
 
-    def natural_resource_name_by_type(type)
-      type_names = { cpu_cores: 'CPU cores', memory_mb: 'Memory (MB)', disk_gb: 'Disk space (GB)' }
-      type_names[type]
+    def natural_resource_name_by_type(resource_type)
+      type_names = { cpu_cores: 'CPU cores', memory_mb: 'Memory', disk_gb: 'Disk space' }
+      key = resource_type.is_a?(String) ? resource_type.to_sym : resource_type
+      raise "No natural name for unknown resource type '#{resource_type}'" unless type_names.key?(key)
+      type_names[key]
     end
 
-    def build_missing_resources_per_host_list(hosts, quota_utilization)
-      # missing_res_per_host := { <host_id>: [<list of to be determined resources>] }
-      # for example: { 1: [ :disk_gb ], 2: [ :cpu_cores, :disk_gb ] }
-      return {} if hosts.empty? || quota_utilization.empty?
+    def units_by_type(resource_type)
+      type_units = {
+        cpu_cores: [
+          { symbol: 'cores', factor: 1 },
+        ],
+        memory_mb: [
+          { symbol: 'MB', factor: 1 },
+          { symbol: 'GB', factor: FACTOR_B_TO_KB },
+          { symbol: 'TB', factor: FACTOR_B_TO_MB },
+        ],
+        disk_gb: [
+          { symbol: 'GB', factor: 1 },
+          { symbol: 'TB', factor: FACTOR_B_TO_KB },
+          { symbol: 'PB', factor: FACTOR_B_TO_MB },
+        ],
+      }
+      key = resource_type.is_a?(String) ? resource_type.to_sym : resource_type
+      raise "No units for unknown resource type '#{resource_type}'" unless type_units.key?(key)
+      type_units[key]
+    end
 
-      missing_res_per_host = hosts.map(&:id).index_with { [] }
-      missing_res_per_host.each_key do |host_id|
-        quota_utilization.each do |resource|
-          missing_res_per_host[host_id] << resource unless resource.nil?
-        end
+    def find_largest_unit(resource_value, units)
+      units.reverse_each do |unit|
+        return unit.values_at(:symbol, :factor) if resource_value >= unit[:factor]
       end
-      missing_res_per_host
+      units[0].values_at(:symbol, :factor)
     end
 
-    def utilization_from_resource_origins(resources, hosts, use_compute_resource: true, use_vm_attributes: true,
-                                          use_facts: true)
-      utilization = resources.each.with_object({}) { |key, hash| hash[key] = 0 }
-      missing_res_per_host = build_missing_resources_per_host_list(hosts, resources)
+    def resource_value_to_string(resource_value, resource_type)
+      (symbol, factor) = find_largest_unit(resource_value, units_by_type(resource_type))
+      unit_applied_value = (resource_value / factor).round(1)
+      format_text = if (unit_applied_value % 1).zero?
+                      '%.0f %s'
+                    else
+                      '%.1f %s'
```

Comment:
>     * In the case of CPU cores, there are no floating point values. So, I would also prefer "10 cores" instead of "10.0  cores"

This is the most convincing for me :)